plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -244,11 +247,36 @@ func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
 }
 
 func (s *SQLServer) gatherServer(server string, query Query, acc telegraf.Accumulator) error {
-	// deferred opening
-	conn, err := sql.Open("mssql", server)
-	if err != nil {
-		return err
+	// deferred connection opening
+	var conn *sql.DB
+
+	if strings.Contains(strings.ToLower(server), strings.ToLower("password=")) {
```

Comment:
I rather like to see a regular expression that makes sure that `password` is the beginning of the parameter and not e.g. part of `o_i_lost_my_password=damn` or `ignorepassword=true`. :-)

Reply:
Addressed the same using regular expression